cup-maven-plugin/src/main/java/de/jflex/plugin/cup/CliCupInvoker.java

```diff
@@ -0,0 +1,100 @@
+package de.jflex.plugin.cup;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import java.io.File;
+import java.util.ArrayList;
+import java_cup.Main;
+import org.apache.maven.plugin.logging.Log;
+
+/** Wrapper around the dirty CUP API. */
+class CliCupInvoker {
+
+  private final Log log;
+
+  CliCupInvoker(Log log) {
+    this.log = log;
+  };
+
+  /**
+   * Invokes CUP.
+   *
+   * @param javaPackage Specify that the parser and sym classes are to be placed in the named
+   *     package. By default, no package specification is put in the generated code (hence the
+   *     classes default to the special "unnamed" package).
+   * @param outputDirectory Directory in which to output the generated Java Parser.
+   * @param parserClassName Output parser and action code into a file (and class) with the given
+   *     name instead of the default of "parser".
+   * @param symClassName Output the symbol constant code into a class with the given name instead of
+   *     the default of "sym".
+   * @param symbolInterface Output the symbol constant code as an interface rather than as a class.
+   * @param cupFileName input specification.
+   */
+  void invoke(
+      String javaPackage,
+      File outputDirectory,
+      String parserClassName,
+      String symClassName,
+      boolean symbolInterface,
+      String cupFileName)
+      throws Exception {
+    // Seriously? cup doesn't have a better API than calling main like on cli!
```

Comment:
It'd be nice to phrase that more neutrally ;-) E.g. `CUP only has the command line as API, so we have to interface there.`

CUP is very old, like JFlex, and it hasn't seen as much active maintenance, so there are a few crufty things that one wouldn't write that way any more these days.

Reply:
Yes, true. I was between surprise and exasperation. Rephrased.

Reply:
And please use **squash** for merging :-)

Reply:
Yes :-)